internal/resources/cloud/resource_private_datasource_connect.go

```diff
@@ -0,0 +1,253 @@
+package cloud
+
+import (
+	"context"
+	"fmt"
+	"slices"
+	"time"
+
+	"github.com/grafana/grafana-com-public-clients/go/gcom"
+	"github.com/grafana/terraform-provider-grafana/v3/internal/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+var (
+	pdcNetworkID = common.NewResourceID(
+		common.StringIDField("region"),
+		common.StringIDField("policyId"),
+	)
+)
+
+func resourcePDCNetwork() *common.Resource {
+	schema := &schema.Resource{
+		Description: `
+* [Official documentation](https://grafana.com/docs/grafana-cloud/connect-externally-hosted/private-data-source-connect/)
+* [API documentation](https://grafana.com/docs/grafana-cloud/developer-resources/api-reference/cloud-api/#create-an-access-policy)
+
+Required access policy scopes:
+
+* accesspolicies:read
+* accesspolicies:write
+* accesspolicies:delete
+`,
+
+		CreateContext: withClient[schema.CreateContextFunc](createPDCNetwork),
+		UpdateContext: withClient[schema.UpdateContextFunc](updatePDCNetwork),
+		DeleteContext: withClient[schema.DeleteContextFunc](deletePDCNetwork),
+		ReadContext:   withClient[schema.ReadContextFunc](readPDCNetwork),
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				Description:  "Region where the API is deployed. Generally where the stack is deployed. Use the region list API to get the list of available regions: https://grafana.com/docs/grafana-cloud/developer-resources/api-reference/cloud-api/#list-regions.",
```

Comment:
I feel like this should be more certain for our users:

```suggestion
				Description:  "The region where your stack is deployed. Use the instances list API to get the region for your instance - use the regionSlug property: https://grafana.com/docs/grafana-cloud/developer-resources/api-reference/cloud-api/#list-stacks",
```

Reply:
Done!